### PR TITLE
Add rspec

### PR DESCRIPTION
**Pull request by LyubovP**

### CODE REVIEW

File: spec/factories/admin_users.rb [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#
+# Indexes
+#
+#  index_admin_users_on_email                 (email) UNIQUE
+#  index_admin_users_on_reset_password_token  (reset_password_token) UNIQUE
+#
 FactoryBot.define do
   factory :admin_user do
```
Comment (rogergraves) [inline spec/factories/admin_users.rb:21]: Missing fields (email, password)

File: spec/factories/categories.rb [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+# Table name: categories
+#
+#  id         :bigint           not null, primary key
+#  name       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
 FactoryBot.define do
   factory :category do
     name { "MyString" }
```
Comment (rogergraves) [inline spec/factories/categories.rb:12]: maybe use `Faker::Company.buzzword`?

File: spec/factories/users.rb [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+#  index_users_on_reset_password_token  (reset_password_token) UNIQUE
+#
 FactoryBot.define do
-  factory :user , aliases: [:doctor, :patient] do
+  factory :patient, class: User do
     email { Faker::Internet.email }
-    password_digest {BCrypt::Password.create("123456")}
-    admin { [UserAdmin::FALSE] }
+    password { Faker::Internet.password }
+    password_confirmation { password }
```
Comment (rogergraves) [inline spec/factories/users.rb:25]: I've found it's easier to set passwords to the same thing for tests, something easy to remember like `Password1!`. That way you don't have to guess what they are in controller tests for example.

File: spec/factories/users.rb [hunk 4]
```diff
@@ -1,7 +1,36 @@
...
+    admin { false}
+    type { "Patient" }
+  end
+  factory :doctor, class: User do
+    email { Faker::Internet.email }
+    password { Faker::Internet.password }
+    password_confirmation { password }
+    admin { false}
+    type { "Doctor" }
```
Comment (rogergraves) [inline spec/factories/users.rb:34]: You might be able to reduce duplication like so:
```ruby
factory :user do
    email { Faker::Internet.email }
    password { 'Password1!' }
    admin { false}
    factory :doctor do
        type { 'Doctor' }
    end
    factory :patient do
       type { 'Patient' }
    end
end
```

**Review by rogergraves — COMMENTED** [review]
For all model tests I suggestion always having these sections:
```ruby
describe 'Factories' do
   it 'profile factory works' do
      expect(profile).to be_valid
   end
end

describe 'Relationships' do
   it { should have_one(:profile) }
   ...
end

describe 'Validations' do
   ...
end

describe 'Scopes' do
   ...
end

describe 'Methods' do
  ...
end
```
Of course you can skip these sections where you don't have anything (like methods and scopes).

**Comment by LyubovP** [comment]
resolved

**Pull request merged**